### DISCUSSION TITLE
Is there any way snoop can identify everything under a WebBrowserControl ?

### DISCUSSION THREAD

**Discussion by ayazhamiya** [snoopwpf/snoopwpf#325] [Q&A] [answered]
I have a WPF application which has in-built browser using WebBrowserControl. I would like to know if snoop can inspect the controls and elements?
For now, I can only see a Canvas tag but nothing more.
<img width="1171" alt="snoopWpfBrowserControl" src="https://user-images.githubusercontent.com/24397824/165115927-6434c1b9-407a-4fe7-9f62-0e9c2b74aec9.png">

**Comment by batzen** [comment, marked as answer]
That's because the rendered elements inside the browser are not WPF elements.
So there is no easy way for Snoop to get those elements.

What would be possible, and relatively easy, is to add some way to open the dev tools of the browser on demand.
Would that fit your needs?

**Reply by ayazhamiya** [reply]
@batzen Thanks for the reply. Yes opening a browser and using dev tools is a good solution. However, the remote debugging must be enabled from the WPF application, right? Otherwise it won't work.

**Reply by batzen** [reply]
Snoop would take care of any steps required to show the dev tools. CefSharp, the component in your sample, provides a simple method to show the dev tools and does not seem to require enabling remote debugging.

**Reply by batzen** [reply]
I created a tracking issue for this.
